src/commands/generate/fromTemplate.ts

```diff
@@ -100,6 +101,15 @@ export default class Template extends Command {
     'map-base-url': Flags.string({
       description: 'Maps all schema references from base url to local folder'
     }),
+    'registry-url': Flags.string({
+      description: 'Specifies the URL of the private registry for fetching templates and dependencies'
+    }),
+    'registry-auth': Flags.string({
+      description: 'The registry username and password encoded with base64, formatted as username:password'
+    }),
+    'registry-token': Flags.string({
+      description: 'The npm registry authentication token, that can be passed instead of base64 encoded username and password'
```

Comment:
i think it will make more sense if we take `registryUsername` and `registryPassword` as input from the user
```suggestion
  'registry-username': Flags.string({
      description: 'The username for the private registry.',
    }),
    'registry-password': Flags.string({
      description: 'The password for the private registry.',
    }),
```

Reply:
Hmm, that was also my initial concern I raised under the issue. Though in the example described [here](https://www.asyncapi.com/docs/tools/generator/using-private-template) we are directly providing the base64 encoded value, wouldn't it make sense to do the same here too?

Reply:
any updates on this?